CMakeLists.txt

```diff
@@ -214,17 +242,31 @@ if(BUILD_DX11 AND WIN32)
         file(TO_CMAKE_PATH ${COMPILED_SHADERS} COMPILED_SHADERS)
     endif()
 
-    list(APPEND LIBRARY_SOURCES ${COMPILED_SHADERS}/BC6HEncode_EncodeBlockCS.inc)
-
     if(NOT USE_PREBUILT_SHADERS)
-        add_custom_command(
-            OUTPUT "${COMPILED_SHADERS}/BC6HEncode_EncodeBlockCS.inc"
-            MAIN_DEPENDENCY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/CompileShaders.cmd"
-            DEPENDS ${SHADER_SOURCES}
-            COMMENT "Generating HLSL shaders..."
-            COMMAND ${CMAKE_COMMAND} -E env CompileShadersOutput="${COMPILED_SHADERS}" CompileShaders.cmd > "${COMPILED_SHADERS}/compileshaders.log"
-            WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders"
-            USES_TERMINAL)
+	    if(MINGW)
+		    file(COPY ${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/BC7Encode.hlsl DESTINATION ${COMPILED_SHADERS})
+			file(COPY ${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/BC6HEncode.hlsl DESTINATION ${COMPILED_SHADERS})
+		
+		    call_mingw_fxc_dx11(${COMPILED_SHADERS} BC7Encode TryMode456CS)	
```

Comment:
If you are going to implement the shader compilation inside the CMake, why make it MinGW only? Just use it always for WIN32 & BUILD_DX11 and rename the .cmake and macros to something more generic: ``hlsl_compile`` perhaps?

> This would mean both the CMake and the .cmd script would have to be kept in sync, but at least for this library there are not that many shaders involved.

Reply:
It's mingw only because I don't have Visual Studio and I would not the where-with-all to use it in a timely manner even if I got a copy today.

Reply:
I have updated the script to support Microsoft's SDK fxc.exe.  I was able to finally test it.  The recent change also calls the CMake script, hlsl_compile.cmake even in non-MINGW environments.  You better test MSVC to be sure that I didn't miss anything.